From 2a77e5b244404b3db143ac261609a9d78309fe73 Mon Sep 17 00:00:00 2001
From: Liviu Ionescu <ilg@livius.net>
Date: Thu, 7 Sep 2023 23:07:00 +0300
Subject: [PATCH] system.h: do not define abort() on mingw

---
 gcc/system.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gcc/system.h b/gcc/system.h
index cf45db3f97e..15c419e79a6 100644
--- a/gcc/system.h
+++ b/gcc/system.h
@@ -792,7 +792,11 @@ extern int vsnprintf (char *, size_t, const char *, va_list);
    should normally call 'internal_error' with a specific message.  */
 extern void fancy_abort (const char *, int, const char *)
 					 ATTRIBUTE_NORETURN ATTRIBUTE_COLD;
+
+#if ! defined(__MINGW32__)
+/* It fails with mingw-w64 10 or later. */
 #define abort() fancy_abort (__FILE__, __LINE__, __FUNCTION__)
+#endif
 
 /* Use gcc_assert(EXPR) to test invariants.  */
 #if ENABLE_ASSERT_CHECKING
-- 
2.37.1 (Apple Git-137.1)

